transform/variant/allele_harvester.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+""" given a genomic location, harvest allele and dependencies """
+import requests
+import urllib
+import logging
+import json
+from bmeg.vertex import Allele
+
+
+TIMEOUT = 30
+
+
+def _myvariantinfo(genome, chromosome, start, end, reference_bases,
+                   alternate_bases, annotations=[]):
+    """ retrieve payload from myvariant.info location query"""
+    if (reference_bases == '' or alternate_bases == '' or
+       reference_bases is None or alternate_bases is None):
+            raise ValueError('reference_bases & alternate_bases must be set')
+    url = hit = None
+    try:
+        dbSNP = None
+        for name in annotations:
+            # "dbSNP_RS:novel"
+            if 'dbSNP_RS' not in name:
+                continue
+            if name == 'dbSNP_RS:novel':
+                continue
+            if name == 'dbSNP_RS:.':
+                continue
+            dbSNP = name.split(':')[1]
+
+        vcf_alternate = alternate_bases
+        if alternate_bases == '-':
+            vcf_alternate = reference_bases
+
+        q = 'chrom:{} AND hg19.start:{} AND hg19.end:{} AND vcf.alt:{}' \
+            .format(chromosome, start, end, vcf_alternate)
+        # logging.debug(genome, chromosome, start, end, alternate_bases)
+        logging.debug(q)
+        url = 'http://myvariant.info/v1/query?q={}' \
+            .format(urllib.parse.quote_plus(q))
+        response = requests.get(url, timeout=TIMEOUT)
+        logging.debug(response)
+        _status_code = response.status_code
+        if _status_code == 200:
+            r = response.json()
+            # return first response
+            for hit in r['hits']:
+                return hit
+
+        # no hit?, lookup by snp if we have it
+        # ["00058da0fb86f362e1504b6ec02d5e4446d9db44","GRCh37","16","397035","397035","C","T",["Variant_Type:SNP","Feature_type:Transcript","Feature:ENST00000262320","dbSNP_RS:rs375097914"]]
+        if dbSNP:
+            url = 'http://myvariant.info/v1/variant/{}'.format(dbSNP)
```

Comment:
Are you aware of https://pypi.org/project/myvariant/ ?

Reply:
Yes, but by explicitly using requests we have options to cache.